src/Microsoft.Diagnostics.EventFlow.Consumers.EventContextConsoleApp/Program.cs

```diff
@@ -0,0 +1,62 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Diagnostics.Correlation.Common;
+using Newtonsoft.Json;
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.EventFlow.Consumers.EventContextConsoleApp
+{
+    public class CorrelationTelemetryInitializer : ITelemetryInitializer
+    {
+        public void Initialize(ITelemetry telemetry)
+        {
+            //add request id to every event
+            var ctx = ContextResolver.GetRequestContext<MyContext>();
+            if (ctx != null)
+            {
+                telemetry.Context.Operation.Id = ctx.CorrelationId;
+            }
+        }
+    }
+
+    class MyContext
+    {
+        public string CorrelationId;
+        public string OtherId;
+
+        public override string ToString()
+        {
+            return JsonConvert.SerializeObject(this);
```

Comment:
If you have just two string properties to serialize, serializing using JsonTextWriter (WriteStartObject, WritePropertyName, WriteValue, WriteEndObject) is much more efficient than JsonConvert.SerializeObject